tests/integration/store-test.js

```diff
@@ -55,6 +55,21 @@ module('Integration - Store', function (hooks) {
     assert.equal(planet.name, 'Earth');
   });
 
+  test('using set to update a record prop on a store fork', async function (assert) {
+    const planet = await store.addRecord({ type: 'planet' });
+    const fork = store.fork();
+    const forkedPlanet = await fork.findRecord('planet', planet.id);
+    forkedPlanet.set('name', 'Earth');
+    await fork.requestQueue.process();
+
+    // Calling process() a second time would make this pass, but was not
+    // necessary in version 0.15.x
+    // await fork.requestQueue.process();
```

Comment:
Does this also work if you call `await fork.syncQueue.process();` here?

In general, both the request and sync queues will need to be flushed to guarantee that a source, even a fork, is "done". 

For this reason, we recommend using async methods to make updates if you need to know exactly when they're complete. Either `forkedPlanet.replaceAttribute('name', 'Earth')` or `forkedPlanet.update(t => t.replaceAttribute('name', 'Earth'))` should do the trick.

Reply:
`await fork.syncQueue.process();` does not help (updated in the test).

Regarding using async methods to make updates... I switched a few places in our app to use them and it does indeed fix the problem, however I feel like there are drawbacks to that being the official solution. One of the biggest drawbacks is that, as an Ember developer, it is very surprising to find that `set` works in some cases but not others. For example it _does_ work if used as listed in the [Updating records](https://github.com/orbitjs/ember-orbit#updating-records) section of the readme, but then does not work if used in a fork. Very surprising.